### PR TITLE
Add validation for file and image size and extension.

### PR DESCRIPTION
**Pull request by RoySegall**
#14

This is the start:
selection_089

### CODE REVIEW

**Comment by RoySegall** [comment]
@amitaibu I've made a change to the API. Now we only have getFieldsInfo instead of the setFieldsInfo.

File: includes/EntityValidateBase.php [hunk 1]
```diff
@@ -79,35 +79,33 @@ public function getEntityType() {
...
+    $instances_info = field_info_instances($this->getEntityType(), $this->getBundle());
 
-    foreach ($fields as $field_name => $info) {
-      // Loading default value of the fields and the instance.
-      $instance_info = field_info_instance($this->entityType, $field_name, $this->bundle);
+    foreach ($instances_info as $instance_info) {
 
       if ($instance_info['required']) {
-        $fields[$field_name]['validators'][] = 'isNotEmpty';
+        // Loading default value of the fields and the instance.
```
Comment (RoySegall) [inline includes/EntityValidateBase.php]: => `Adding validation of empty fields for required fields`

File: includes/EntityValidateBase.php [hunk 2]
```diff
@@ -79,35 +79,33 @@ public function getEntityType() {
...
+        $fields[$instance_info['field_name']]['validators'][] = 'isNotEmpty';
+      }
+
+      $field_info = field_info_field($instance_info['field_name']);
+
+      if ($field_info['type'] == 'image') {
+        $fields[$instance_info['field_name']]['validators'][] = 'validateImageField';
```
Comment (RoySegall) [inline includes/EntityValidateBase.php]: Add comment like `Add validate method for image fields`

File: tests/EntityValidatorTestCase.test [hunk 3]
```diff
@@ -7,6 +7,9 @@
 
 class EntityValidatorTestCase extends DrupalWebTestCase {
 
+  private $instance;
```
Comment (RoySegall) [inline tests/EntityValidatorTestCase.test]: Add the comments.

File: tests/EntityValidatorTestCase.test [hunk 4]
```diff
@@ -16,19 +19,74 @@ class EntityValidatorTestCase extends DrupalWebTestCase {
...
+        'min_resolution' => '41X21',
+        'alt_field' => TRUE,
+        'title_field' => '',
+        'default_image' => '',
+      ),
+
+      'widget' => array(),
+    );
+
+    if (!field_info_instance('node', 'field_image', 'article')) {
```
Comment (amitaibu) [inline tests/EntityValidatorTestCase.test]: in a test there's no need for this check.

**Comment by RoySegall** [comment]
@amitaibu I fixed the tests.

**Comment by amitaibu** [comment]
needs re-roll

**Comment by RoySegall** [comment]
Re rolled.

File: includes/EntityValidateBase.php [hunk 5]
```diff
@@ -263,4 +262,59 @@ public function isValidValue($field_name, $value, $type) {
       $this->setError($field_name, 'The value @value is invalid for the field @field.', $params);
     }
   }
+
+  /**
+   * Validate the field image: Check the image is the correct size.
+   */
+  public function validateImageField($field_name, $value) {
+    $info = field_info_instance($this->getEntityType(), $field_name, $this->getBundle());
```
Comment (amitaibu) [inline includes/EntityValidateBase.php:284]: => instance_info

File: includes/EntityValidateBase.php [hunk 6]
```diff
@@ -79,35 +79,34 @@ public function getEntityType() {
...
+    $instances_info = field_info_instances($this->getEntityType(), $this->getBundle());
 
-    foreach ($fields as $field_name => $info) {
-      // Loading default value of the fields and the instance.
-      $instance_info = field_info_instance($this->entityType, $field_name, $this->bundle);
+    foreach ($instances_info as $instance_info) {
 
       if ($instance_info['required']) {
-        $fields[$field_name]['validators'][] = 'isNotEmpty';
+        // Validate that required fields will not be empty.
```
Comment (amitaibu) [inline includes/EntityValidateBase.php]: => Validate field is not empty.

File: includes/EntityValidateBase.php [hunk 7]
```diff
@@ -79,35 +79,34 @@ public function getEntityType() {
...
+        $fields[$instance_info['field_name']]['validators'][] = 'isNotEmpty';
+      }
+
+      $field_info = field_info_field($instance_info['field_name']);
+
+      if ($field_info['type'] == 'image') {
```
Comment (amitaibu) [inline includes/EntityValidateBase.php:107]: we should probably also add another validation for file/image -- to validate the extension.

File: modules/entity_validator_example/plugins/validator/node/article/EntityValidatorExampleArticleValidator.class.php [hunk 8]
```diff
@@ -10,8 +10,8 @@ class EntityValidatorExampleArticleValidator extends EntityValidateBase {
   /**
    * Overrides NodeValidate::getFieldsInfo().
    */
-  public function setFieldsInfo() {
-    $fields = parent::setFieldsInfo();
+  public function getFieldsInfo() {
```
Comment (amitaibu) [inline modules/entity_validator_example/plugins/validator/node/article/EntityValidatorExampleArticleValidator.class.php:13]: PHPdoc seems wrong (NodeValidate)

File: includes/EntityValidateBase.php [hunk 9]
```diff
@@ -263,4 +262,59 @@ public function isValidValue($field_name, $value, $type) {
       $this->setError($field_name, 'The value @value is invalid for the field @field.', $params);
     }
   }
+
+  /**
+   * Validate the field image: Check the image is the correct size.
+   */
+  public function validateImageField($field_name, $value) {
```
Comment (amitaibu) [inline includes/EntityValidateBase.php]: missing params

File: includes/EntityValidateBase.php (hunk 7 above)
Comment (RoySegall) [inline includes/EntityValidateBase.php:107]: Added.

**Comment by RoySegall** [comment]
@amitaibu I've added the tests. Can you please review?

**Comment by amitaibu** [comment]
Ok, please merge

**Comment by amitaibu** [comment]
Actually, I see an error in the extension name. 

angularjs_form___site-install
Ok, please merge

**Comment by amitaibu** [comment]
and also an error when not uploading a file

angularjs_form___site-install

**Comment by amitaibu** [comment]
- please add spaces before the `()`

**Comment by RoySegall** [comment]
selection_006

**Comment by amitaibu** [comment]
Please check the failing tests

**Comment by RoySegall** [comment]
The tests now working.

File: includes/EntityValidateBase.php [hunk 10]
```diff
@@ -79,35 +79,39 @@ public function getEntityType() {
...
+      $field_info = field_info_field($instance_info['field_name']);
+
+      if ($field_info['type'] == 'image') {
+        // Validate the image dimensions.
+        $fields[$instance_info['field_name']]['validators'][] = 'validateImageSize';
+      }
+
+      if (in_array($field_info['type'], array('image', 'file'))) {
+        // Validate the file type.
+        $fields[$instance_info['field_name']]['validators'][] = 'validateFileExtension';
```
Comment (amitaibu) [inline includes/EntityValidateBase.php:114]: as a follow up lets add a `validateFileSize`

**Comment by amitaibu** [comment]
Ok, please merge.

**Pull request closed without merging**